packages/pwa-kit-react-sdk/src/ssr/universal/components/route-component/index.js

```diff
@@ -396,30 +396,33 @@ export const routeComponent = (Wrapped, isPage, locals) => {
 }
 
 /**
+ * TODO: update this comment
  * Wrap all the components found in the application's route config with the
  * route-component HOC so that they all support `getProps` methods server-side
  * and client-side in the same way.
  *
  * @private
  */
-export const getRoutes = (locals = {}) => {
-    let _routes = routes
+export const getAllRoutes = async (locals = {}) => {
     const {applicationExtensions = []} = locals
-    if (typeof routes === 'function') {
-        _routes = routes()
-    }
-
-    // Call the `extendRoutes` function for all the Application Extensions.
-    applicationExtensions.forEach((applicationExtension) => {
-        _routes = applicationExtension.extendRoutes(_routes)
-    })
+    const extensionRoutes = (
+        await Promise.all(
+            applicationExtensions.map((extension) =>
+                extension.getRoutesAsync
+                    ? extension.getRoutesAsync({locals})
+                    : extension.getRoutes()
+            )
+        )
```

Comment:
I see that the async method is taking priority over getRoutes, which I like. But there is technically nothing stopping developers from defining both. In the absence of an eslint rule, do you think we should print a warning?

Reply:
Hmm, `getRoutes` has a default implementation (which returns an empty array). So if an extension does not define `getRoutes` themselves, the method still exists.

In other words, if the extension only defines `getRoutesAsync`, it will still inherit `getRoutes` from the parent class. So I'm not sure how to detect and warn the developers accordingly.